### ISSUE TITLE
xip.io hostname not generated

### ISSUE THREAD

**Issue by phertweck** [rancher/rancher#14960]
**Rancher versions:**
rancher/server or rancher/rancher: 2.0.6
rancher/agent or rancher/rancher-agent: 2.0.6

**Docker version: (`docker version`,`docker info` preferred)**
 18.06.0-ce

**Operating system and kernel: (`cat /etc/os-release`, `uname -r` preferred)**
Debian GNU/Linux 9 (stretch)

**Type/provider of hosts: (VirtualBox/Bare-metal/AWS/GCE/DO)**
Bare-metal

**Setup details: (single node rancher vs. HA rancher, internal DB vs. external DB)**
HA rancher with external HAProxy

**Steps to Reproduce:**
* Adding new ingress and selecting `Automatically generate a .xip.io hostname`

**Results:**
* Ingress is created with `xip.io` hostname (no subdomain is added)

**Comment by loganhz** [comment]
I guess it took around 1 minute to generate the subdomain.

**Comment by phertweck** [comment]
@loganhz Thank you for this information. I waited several minutes, but no sudomain is created, so the service is available on the root doamin name.

On an other rancher instance (single node) the subdomain is created immediately.

**Comment by dnauck** [comment]
I've the same problem with a "custom cluster". Looks like the generator does not know the ip of the nodes ?

**Comment by phertweck** [comment]
I also tried to set the ingress-ip-domain to `<myip>.xio.io` so there shouldn't be any problem of getting the nodes ip.

**Comment by adrianocr** [comment]
I too have this issue. A .xip.io hostname is never generated. Waited over an hour to be sure.

Are there logs somewhere I can check to specify the error?

**Comment by rreilly-edr** [comment]
+1 same issue

**Comment by augustynski-lukasz** [comment]
I had this issue too. A .xip.io hostname have never been generated. After change the ranger www port from  80/443 to other, the problem disappeared.

**Comment by milefork** [comment]
I've the same issue too.
My Setup are 2 bare metal servers. Maybe the bare metal is the point?

Another ref:
[https://forums.rancher.com/t/nginx-ingres-auto-xip-io-generation-doesnt-work-with-custom-cloud-provider-enabled/11605](url)

**Comment by myclau** [comment]
I create the cluster again and in the cluster.yml change all node address using ip instead of host name. and xip.io can generate successfully

**Comment by pkeshab** [comment]
Hello guys,

I am using ingress to expose sample app to the internet according to Rancher 2.0 documentation. Amazon ec2 instance is used as 1 node customized cluster and while using ingress the private ip of ec2 is converted to public dns with xip.io but when I click the link the page never opens.
Error is:
test.default.<ec2-private-ip>.xip.io takes too long to respond
I guess this should be simple but not able to overcome it. Can anyone please assist me?

**Comment by myclau** [comment]
> Hello guys,
> 
> I am using ingress to expose sample app to the internet according to Rancher 2.0 documentation. Amazon ec2 instance is used as 1 node customized cluster and while using ingress the private ip of ec2 is converted to public dns with xip.io but when I click the link the page never opens.
> Error is:
> test.default..xip.io takes too long to respond
> I guess this should be simple but not able to overcome it. Can anyone please assist me?

From http://xip.io
```
               10.0.0.1.xip.io   resolves to   10.0.0.1
      www.10.0.0.1.xip.io   resolves to   10.0.0.1
   mysite.10.0.0.1.xip.io   resolves to   10.0.0.1
  foo.bar.10.0.0.1.xip.io   resolves to   10.0.0.1
```
so that for the machine you go to test.default.XXX.xip.io you need to have access to XXX first

my be you want to read this:
https://stackoverflow.com/questions/26818456/amazon-ec2-not-working-when-accessing-through-public-ip

**Comment by pkeshab** [comment]
Hi myclau,
I am not facing problems with ec2 public ip. To make it clear, I ran a sample rancher/hello-world work load in rancher 2.0 ( its one node cluster is an aws-ec2 ubuntu instance). After that to expose to external world, I selected Load Balancing option in the rancher and xip.io option to create hostname. After that documentation says, we are provide with the link with xip.io and when we click the link the application should open in the browser as hello rancher sample application. But here I am not able to load the link provided by xip.io hostname.
Thank you

**Comment by Amos-85** [comment]
same issue.
I'm running on HA installation / rancher v2.1.4
ingress created without the hostname subdomain.

**Comment by deniseschannon** [comment]
if anyone is hitting htis on a more modern version, please open a new ticket.

**Comment by gabrielcalderon** [comment]
@deniseschannon This is still happening in a different way, I'm running bare-metal with 8443:443 and 8080:80 on Rancher 2.2.2

I'm going to create a separate bug ticket.

**Comment by azubizarretap** [comment]
I am facing this on 2.2.2 on vsphere. Any clues?

**Comment by Aisuko** [comment]
Same issue on rancher v2.1.4. But on v2.2.2 working normally.

**Comment by Aisuko** [comment]
> https://forums.rancher.com/t/nginx-ingres-auto-xip-io-generation-doesnt-work-with-custom-cloud-provider-enabled/11605

The address removed?

**Comment by hbudde** [comment]
The problem still exists. I have created a cluster with rke (v0.2.8) and I used an internal DNS name like webmaster01.example.local for the address in the cluster.yml. The underlying OS can resolve the DNS name and also the cluster was working as expected. In Rancher (v2.2.8) I can see the DNS / IP in the nodes section, but the host part of the ingress is not resolved. If I change the address in the cluster.yml to the IP of the nodes, it is working as expected and the ingress resolves the host part.
